### ISSUE TITLE
java.lang.NumberFormatException

### ISSUE THREAD

**Issue by bpcw** [semver4j/semver4j#208]
Hi,

starting from semver4j versions beyond 1.0.1, the following program does not work any more and results in a java.lang.NumberFormatException

```
import org.semver4j.Semver;

public class SemVerTest {

    public static void main(String[] args) {

        Semver versionOld = new Semver("1.24.1-A-20240111143214+a35df53040615b65742729058655a85fa98b84d3");
        Semver versionNew = new Semver("1.24.1-A-20240117112233+bb1b81e5cf1a3ae8ee736590454d023926733417");

        System.out.println(versionOld.compareTo(versionNew));
    }
}
```

Essentially, the program compares two fictional build versions (1.24.1-A) which only differ by a build timestamp and a commit id. 
It seems the compare should not crash when comparing these versions.

BR
bpcw

**Comment by piotrooo** [comment]
What version are you currently using?

**Comment by bpcw** [comment]
```
<dependency>
    <groupId>org.semver4j</groupId>
    <artifactId>semver4j</artifactId>
    <version>5.2.2</version>
</dependency>
```

**Comment by piotrooo** [comment]
Seems to be a bug in the comparator.

```
java.lang.NumberFormatException: For input string: "20240111143214"

	at java.base/java.lang.NumberFormatException.forInputString(NumberFormatException.java:67)
	at java.base/java.lang.Integer.parseInt(Integer.java:668)
	at java.base/java.lang.Integer.parseInt(Integer.java:786)
	at org.semver4j.internal.Comparator.compareIdentifiers(Comparator.java:92)
	at org.semver4j.internal.Comparator.preReleaseCompare(Comparator.java:72)
	at org.semver4j.internal.Comparator.compareTo(Comparator.java:25)
	at org.semver4j.Semver.compareTo(Semver.java:328)
	at org.semver4j.SemverTest.shouldName(SemverTest.java:46)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
```

**Comment by bpcw** [comment]
Yep. it worked up to (and including) semver4j 1.0.1.
All following versions as far as my tests go, up to the latest 5.2.2 version, suffer from this issue.

**Comment by theghost5800** [comment]
When we can expect a fix for this issue?

**Comment by severn-everett** [comment]
I'll fix it shortly.

**Comment by piotrooo** [comment]
Many thans @severn-everett!
Fixed in [v5.2.3](https://github.com/semver4j/semver4j/releases/tag/v5.2.3)
